### ISSUE TITLE
Use a registry object to store all model wrapper funcs

### ISSUE THREAD

**Issue by ivan-lazarevich** [Deeplite/deeplite-torch-zoo#99]
- Move the [Registry class implementation](https://github1s.com/Deeplite/deeplite-torch-zoo/blob/HEAD/deeplite_torch_zoo/src/objectdetection/utils/registry.py#L1-L2) to a higher level (e.g. deeplite_torch_zoo.utils) 
 - Use the registry object to register model wrapper funcs and use that in `list_all_models` and `get_model_by_name` instead of `globals()`
 - Extend registry impl to have two keys: `model_name` and `dataset` and register with those like
    ```
    @WRAPPER_REGISTRY.register(model_name='mobilenetv1', dataset='cifar100')
    def mobilenetv1_cifar100_wrapper():
    ```
    inside it can be a dict with 2-element tuples as keys

cc @ahmed-deeplite

**Comment by shahrukh-dl** [comment]
Sample outout

Available Models          | Trained on datasets
--------------------------------------------------------------------------------
mb1_ssd                   | voc20
mb2_ssd_lite              | voc20, voc_1, voc_2, subcls
vgg16_ssd                 | voc_20, wider_face
yolo3                     | voc_20, voc_1, voc_2, lisa_11
yolo4                     | 
yolo4s                    | voc_20
yolo4m                    | voc_20, lisa_11
yolo4l                    | voc_20
yolo4x                    | voc_20
yolo4l_leaky              | voc_20
yolo5                     | 
yolo5_6                   | 
yolo5_6n                  | person_detection_1, voc_20, coco_80
yolo5_6s                  | person_detection_1, voc_20, coco_80
yolo5_6n_relu             | person_detection_1
yolo5_6s_relu             | person_detection_1
yolo5s                    | voc_20, coco_80
yolo5m                    | voc_20, voc_24, wider_face_8, coco_80
yolo5l                    | voc_20, voc_24, wider_face_8, coco_80
yolo5x                    | voc_20, coco_80
yolo5_6m                  | voc_20, coco_80
mb2_ssd                   | coco_gm_6, voc20, coco_80
fasterrcnn_resnet50_fpn   | coco_80
resnet18_ssd              | voc_20
resnet34_ssd              | voc_20
resnet50_ssd              | voc20
deeplab_mobilenet         | voc_20
fcn32                     | voc20
unet                      | carvana
unet_enc_dec              | 
unet_scse_resnet18        | voc_20, voc_1, voc_2, carvana
densenet121               | cifar100, imagenet
densenet161               | imagenet
densenet169               | imagenet

**Comment by ivan-lazarevich** [comment]
Addressed in #113
